ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Peras/Weight.hs

```diff
@@ -57,35 +127,120 @@ addToPerasWeightSnapshot ::
-removeFromPerasWeightSnapshot pt (PerasWeight weight) =
-  PerasWeightSnapshot . Map.update subtractWeight pt . getPerasWeightSnapshot
+prunePerasWeightSnapshot slot =
+  PerasWeightSnapshot . Map.dropWhileAntitone isTooOld . getPerasWeightSnapshot
  where
-  subtractWeight :: PerasWeight -> Maybe PerasWeight
-  subtractWeight (PerasWeight w)
-    | w > weight = Just $ PerasWeight (w - weight)
-    | otherwise = Nothing
+  isTooOld :: Point blk -> Bool
+  isTooOld pt = pointSlot pt < NotOrigin slot
 
-boostedWeightForPoint ::
+-- | Get the weight boost for a point, or @'mempty' :: 'PerasWeight'@ otherwise.
+--
+-- >>> :{
+-- weights :: [(Point Blk, PerasWeight)]
+-- weights =
+--   [ (BlockPoint 2 "foo", PerasWeight 2)
+--   , (GenesisPoint,       PerasWeight 3)
+--   , (BlockPoint 3 "bar", PerasWeight 2)
+--   , (BlockPoint 2 "foo", PerasWeight 2)
+--   ]
+-- :}
+--
+-- >>> snap = mkPerasWeightSnapshot weights
+--
+-- >>> weightBoostForPoint snap (BlockPoint 2 "foo")
+-- PerasWeight 4
+--
+-- >>> weightBoostForPoint snap (BlockPoint 2 "baz")
+-- PerasWeight 0
+weightBoostForPoint ::
   forall blk.
   StandardHash blk =>
   PerasWeightSnapshot blk -> Point blk -> PerasWeight
-boostedWeightForPoint (PerasWeightSnapshot weightByPoint) pt =
+weightBoostForPoint (PerasWeightSnapshot weightByPoint) pt =
   Map.findWithDefault mempty pt weightByPoint
 
-boostedWeightForFragment ::
+-- | Get the weight boost for a fragment, ie the sum of all
+-- 'weightBoostForPoint' for all points on the fragment (excluding the anchor).
+--
+-- Note that this quantity is relative to the anchor of the fragment, so it
+-- should only be compared against other fragments with the same anchor.
+--
+-- >>> :{
+-- weights :: [(Point Blk, PerasWeight)]
+-- weights =
+--   [ (BlockPoint 2 "foo", PerasWeight 2)
+--   , (GenesisPoint,       PerasWeight 3)
+--   , (BlockPoint 3 "bar", PerasWeight 2)
+--   , (BlockPoint 2 "foo", PerasWeight 2)
+--   ]
+-- :}
+--
+-- >>> :{
+-- snap = mkPerasWeightSnapshot weights
+-- foo = HeaderFields (SlotNo 2) (BlockNo 1) "foo"
+-- bar = HeaderFields (SlotNo 3) (BlockNo 2) "bar"
+-- frag0 :: AnchoredFragment (HeaderFields Blk)
+-- frag0 = Empty AnchorGenesis :> foo :> bar
+-- :}
+--
+-- >>> weightBoostForFragment snap frag0
+-- PerasWeight 6
+--
+-- Only keeping the last block from @frag0@:
+--
+-- >>> frag1 = AF.anchorNewest 1 frag0
+-- >>> weightBoostForFragment snap frag1
+-- PerasWeight 2
+--
+-- Dropping the head from @frag0@, and instead adding an unboosted point:
+--
+-- >>> frag2 = AF.dropNewest 1 frag0 :> HeaderFields (SlotNo 4) (BlockNo 2) "baz"
+-- >>> weightBoostForFragment snap frag2
+-- PerasWeight 4
+weightBoostForFragment ::
   forall blk h.
-  (HasHeader blk, HasHeader h, HeaderHash blk ~ HeaderHash h) =>
+  (StandardHash blk, HasHeader h, HeaderHash blk ~ HeaderHash h) =>
   PerasWeightSnapshot blk ->
   AnchoredFragment h ->
   PerasWeight
-boostedWeightForFragment weightSnap frag =
+weightBoostForFragment weightSnap frag =
```

Comment:
What about `weightBoostOf` instead of `weightBoostFor`? For me, "of" indicates an existing property, while "for" may imply that it hasn't been applied/set yet. 

(same for above with `weightBoost{Of/For}Point`